benchmarks/asv.conf.json

```diff
@@ -113,15 +113,15 @@
     "include": [
         // minimum supported versions
         {
-            "python": "3.7",
+            "python": "3.12.1",
```

Comment:
@rajeshkotaru, as suggested by the comment on line 114, these versions are supposed to match pvlib's minimum requirements.  Currently, pvlib requires at least python 3.7 and the following minimum dependency versions:

https://github.com/pvlib/pvlib-python/blob/e92ee94be24c2a74375d679f6bc420a5b3202bbd/setup.py#L35-L39

So there is a reason that these versions are the way that they are, and increasing them to more recent versions isn't just a simple update, but rather a change to our strategy to doing benchmarking.  I don't think we want to make this change without discussing that strategy change first.